### PR TITLE
[SDPA] Performance improvment to cumulative seq len

### PR DESCRIPTION
**Pull request by drisspg**
# Summary
Performance improvement to calculating metadata needed for gluing in nested tensors to fused kernels.

cc @ngimel

### CODE REVIEW

**Comment by drisspg** [comment]
Benchmark from benchmark/transformer/sdp.py

Benchmarks at 50% padding before change:
|max_sequence_len|num_heads|embed_dimension|pt_time            |cp_time            |speedup           |dtype        |
|----------------|---------|---------------|-------------------|-------------------|------------------|-------------|
|64              |2        |1024           |0.2842316818237305 |0.2992025566101074 |0.9499640813367596|torch.float16|
|128             |2        |1024           |0.46704639434814454|0.37438465118408204|1.247504118747917 |torch.float16|
|256             |2        |1024           |0.8568831634521484 |0.47581184387207026|1.8008865783562449|torch.float16|
|64              |4        |1024           |0.2817945671081543 |0.2992127990722656 |0.9417864743148756|torch.float16|
|128             |4        |1024           |0.4401561737060547 |0.3316326522827148 |1.3272401576755   |torch.float16|
|256             |4        |1024           |0.8648294067382813 |0.45923328399658203|1.8832028010075987|torch.float16|
|64              |8        |1024           |0.27967487335205077|0.29249536514282226|0.9561685642966978|torch.float16|
|128             |8        |1024           |0.4856217575073242 |0.32535552978515625|1.4925879939031539|torch.float16|
|256             |8        |1024           |1.0080665588378905 |0.4530790328979492 |2.2249243192521466|torch.float16|
|64              |16       |1024           |0.273960952758789  |0.2953830337524414 |0.927476941645856 |torch.float16|
|128             |16       |1024           |0.5439180755615235 |0.3308339309692383 |1.6440818931964336|torch.float16|
|256             |16       |1024           |1.308467254638672  |0.45964286804199217|2.8467041383945344|torch.float16|
|64              |32       |1024           |0.3299737548828125 |0.2972159957885742 |1.110215330124899 |torch.float16|
|128             |32       |1024           |0.7490560150146485 |0.3324620819091797 |2.2530569823576805|torch.float16|
|256             |32       |1024           |1.9138560485839846 |0.5307187271118164 |3.606158876283174 |torch.float16|


Benchmark at 50% padding after change:
|max_sequence_len|num_heads|embed_dimension|pt_time            |cp_time            |speedup           |dtype        |
|----------------|---------|---------------|-------------------|-------------------|------------------|-------------|
| 64             |2        |1024           |0.2846208000183106 |0.29784063339233396|0.9556144061894691|torch.float16|
|128             |2        |1024           |0.46267391204833985|0.37412864685058594|1.236670637073979 |torch.float16|
|256             |2        |1024           |0.7856230163574219 |0.4815359878540039 |1.6314938782843653|torch.float16|
|64              |4        |1024           |0.2869964790344239 |0.30248960494995114|0.9487812947552671|torch.float16|
|128             |4        |1024           |0.4410367965698242 |0.3292364883422852 |1.3395744766640436|torch.float16|
|256             |4        |1024           |0.8651468658447266 |0.4607795333862305 |1.8775722512821569|torch.float16|
|64              |8        |1024           |0.3296460723876953 |0.30682111740112306|1.0743917341150022|torch.float16|
|128             |8        |1024           |0.4831232070922852 |0.3293183898925781 |1.4670398675575858|torch.float16|
|256             |8        |1024           |1.0087628936767579 |0.4537446212768555 |2.2231952652971594|torch.float16|
|64              |16       |1024           |0.276551685333252  |0.2946969604492187 |0.9384273421473124|torch.float16|
|128             |16       |1024           |0.5787545776367188 |0.3301580810546875 |1.7529620228827707|torch.float16|
|256             |16       |1024           |1.3093785095214845 |0.46309375762939453|2.8274587768668566|torch.float16|
|64              |32       |1024           |0.3298303985595703 |0.29871103286743167|1.104178829263228 |torch.float16|
|128             |32       |1024           |0.747540512084961  |0.32730110168457033|2.283953546863975 |torch.float16|
|256             |32       |1024           |1.9139379882812502 |0.5186457443237306 |3.6902606629441466|torch.float16|

**Comment by drisspg** [comment]
Need to reduce size to reduce variance in runs

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp [hunk 1]
```diff
@@ -274,7 +275,7 @@ std::tuple<Tensor, int64_t> cumulative_and_max_seq_len(Tensor qkv) {
   // Send to GPU, this is pretty light weight calc for normal batch size
   // but maybe this needs to be on gpu
   cumulative_seqlen = cumulative_seqlen.to(TensorOptions().device(at::kCUDA));
```
Comment (cpuhrsch) [inline aten/src/ATen/native/nested/cuda/NestedTensorTransformerFunctions.cpp:277]: Could make this an async transfer via `non_blocking=true`

**Pull request closed without merging**